components/brave_wallet/browser/brave_wallet_provider_impl.cc

```diff
@@ -648,7 +665,12 @@ void BraveWalletProviderImpl::Locked() {
 }
 
 void BraveWalletProviderImpl::Unlocked() {
-  UpdateKnownAccounts();
+  if (pending_unlock_callback_) {
+    RequestEthereumPermissions(std::move(pending_unlock_callback_));
+    pending_unlock_callback_ = RequestEthereumPermissionsCallback();
```

Comment:
`pending_unlock_callback_` naming is confusing, I was a bit confused at first and didn't realize it's the callback for RequestEthereumPermissions, maybe pending_request_ethereum_permissions_callback_? 
And could you explain to me why do we need to construct a callback and assign to this member when the stored callback is used?

Reply:
Fixed in f466d97e0c79518206aab34f1e3793c5cf861930